easybuild/framework/easyconfig/types.py

```diff
@@ -124,20 +127,59 @@ def to_toolchain(tcspec):
 
     if isinstance(tcspec, list):
         if len(tcspec) == 2:
-            res = {'name': tcspec[0].strip(), 'version': tcspec[1].strip()}
+            if convert_to == TYPE_TUPLE:
```

Comment:
if you rework `to_depenency` and no longer need this case, kill it with fire (and maybe rename this function back to `to_name_version_dict` ;-))